iceoryx2-ffi/cxx/tests/src/service_publish_subscribe_tests.cpp

```diff
@@ -943,9 +986,142 @@ TYPED_TEST(ServicePublishSubscribeTest,
+                          .template user_header<CustomHeader>()
+                          .create();
+
+    struct CustomHeader {
+        uint64_t a;
+        uint8_t b;
+    };
+    auto sut_open = node.service_builder(service_name)
+                        .template publish_subscribe<uint8_t>()
+                        .template user_header<CustomHeader>()
+                        .open();
+    ASSERT_TRUE(sut_open.has_error());
+}
+
+TYPED_TEST(ServicePublishSubscribeTest,
+           opening_existing_service_with_same_header_but_different_user_header_type_name_fails) {
+    constexpr ServiceType SERVICE_TYPE = TestFixture::TYPE;
+    const auto service_name = iox2_testing::generate_service_name();
+
+    auto node = NodeBuilder().create<SERVICE_TYPE>().expect("");
+    auto sut_create = node.service_builder(service_name)
+                          .template publish_subscribe<uint8_t>()
+                          .template user_header<CustomHeader>()
+                          .create();
+
+    auto sut_open = node.service_builder(service_name)
+                        .template publish_subscribe<uint8_t>()
+                        .template user_header<other::CustomHeader>()
+                        .open();
+    ASSERT_TRUE(sut_open.has_error());
+}
+
+TYPED_TEST(ServicePublishSubscribeTest, opening_existing_service_with_same_header_type_but_different_size) {
+    constexpr ServiceType SERVICE_TYPE = TestFixture::TYPE;
+    const auto service_name = iox2_testing::generate_service_name();
+
+    auto node = NodeBuilder().create<SERVICE_TYPE>().expect("");
+    auto sut_create = node.service_builder(service_name)
+                          .template publish_subscribe<uint8_t>()
+                          .template user_header<CustomHeader>()
+                          .create();
+
+    auto sut_open = node.service_builder(service_name)
+                        .template publish_subscribe<uint8_t>()
+                        .template user_header<CustomHeaderWithSameTypeNameButDifferentSize>()
+                        .open();
+    ASSERT_TRUE(sut_open.has_error());
+}
+
+TYPED_TEST(ServicePublishSubscribeTest, opening_existing_service_with_same_header_type_but_different_alignment) {
+    constexpr ServiceType SERVICE_TYPE = TestFixture::TYPE;
+    const auto service_name = iox2_testing::generate_service_name();
+
+    auto node = NodeBuilder().create<SERVICE_TYPE>().expect("");
+    auto sut_create = node.service_builder(service_name)
+                          .template publish_subscribe<uint8_t>()
+                          .template user_header<CustomHeader>()
+                          .create();
+
+    auto sut_open = node.service_builder(service_name)
+                        .template publish_subscribe<uint8_t>()
+                        .template user_header<CustomHeaderWithSameTypeNameButDifferentAlignment>()
+                        .open();
     ASSERT_TRUE(sut_open.has_error());
```

Comment:
Two little nitpicks.
1. Could you also state in the test name what outcome you expect. In this case, please add the suffix `_fails` so that someone else understands that this should not work.
2. Could you also check that here the correct error enum value is returned. I think it would be something like incompatible message type or so.

Could you apply this also to the other tests. And then we get it merged!